bliss/simulator/simulated_dataset.py

```diff
@@ -52,17 +79,45 @@ def _apply_noise(images_mean):
         images += images_mean
         return images
 
-    def simulate_image(self, tile_catalog: TileCatalog) -> Tuple[Tensor, Tensor]:
-        images = self.image_decoder.render_images(tile_catalog)
-        background = self.background.sample(images.shape)
+    def simulate_image(self, tile_catalog: TileCatalog, rcf_indices=None) -> Tuple[Tensor, Tensor]:
+        """Simulate a batch of images.
+
+        Args:
+            tile_catalog (TileCatalog): The TileCatalog to render.
+            rcf_indices: Indices of rcfs in self.rcf_list. Defaults to None.
+
+        Returns:
+            Tuple[Tensor, Tensor]: tuple of images and backgrounds
+        """
+        rcf = self.rcf_list[rcf_indices] if rcf_indices is not None else None
+        images = self.image_decoder.render_images(tile_catalog, rcf)
+        background = self.background.sample(images.shape, rcf_indices=rcf_indices)  # type: ignore
         images += background
         images = self._apply_noise(images)
         return images, background
 
-    def get_batch(self) -> Dict:
+    def get_batch(self, independent=False) -> Dict:
+        """Get a batch of simulated images.
+
+        The images are simulated by first generating a tile catalog from the prior, followed
+        by choosing a random background and PSF and using those to generate the image. By default,
+        the same row, camcol, and field combination is used for the background, PSF, and flux ratios
+        of a single simulated image.
+
+        Args:
+            independent (bool, optional): Use different RCF combinations for background, psf, and
+                flux ratios for each image. Defaults to False.
+
+        Returns:
+            A dictionary of the simulated TileCatalog, and (batch_size, bands, height, width)
+            tensors for images and background.
+        """
+        rcfs, rcf_indices = (
+            self._get_random_rcf(self.image_prior.batch_size) if not independent else None
+        )
         with torch.no_grad():
-            tile_catalog = self.image_prior.sample_prior()
-            images, background = self.simulate_image(tile_catalog)
+            tile_catalog = self.image_prior.sample_prior(rcfs)  # TODO: add rcf
```

Comment:
I don't think I understand the TODO comment. Maybe open a github issue instead?

Reply:
Forgot to remove this, it was just a reminder to pass rcf to sample_prior. Will be fixed in the next commit.